acceptance/openstack/sharedfilesystems/v2/shares.go

```diff
@@ -0,0 +1,63 @@
+package v2
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	"github.com/gophercloud/gophercloud/openstack/sharedfilesystems/v2/shares"
+	"testing"
+)
+
+// CreateShare will create a share with a name, and a size of 1Gb. An
+// error will be returned if the share could not be created
+func CreateShare(t *testing.T, client *gophercloud.ServiceClient) (*shares.Share, error) {
+	if testing.Short() {
+		t.Skip("Skipping test that requres share creation in short mode.")
+	}
+
+	choices, err := clients.AcceptanceTestChoicesFromEnv()
+	if err != nil {
+		t.Fatalf("Unable to fetch environment information")
+	}
+
+	t.Logf("Share network id %s", choices.ShareNetworkID)
+	createOpts := shares.CreateOpts{
+		Size:           1,
+		Name:           "My Test Share",
+		ShareProto:     "NFS",
+		ShareNetworkID: choices.ShareNetworkID,
+	}
+
+	share, err := shares.Create(client, createOpts).Extract()
+	if err != nil {
+		return share, err
+	}
+
+	err = shares.WaitForStatus(client, share.ID, "available", 60)
+	if err != nil {
+		return share, err
+	}
+
+	return share, nil
+}
+
+// DeleteShare will delete a share. A fatal error will occur if the share
+// failed to be deleted. This works best when used as a deferred function.
+func DeleteShare(t *testing.T, client *gophercloud.ServiceClient, share *shares.Share) {
+	err := shares.Delete(client, share.ID).ExtractErr()
+	if err != nil {
+		t.Fatalf("Unable to delete share %s: %v", share.ID, err)
+	}
+
+	t.Logf("Deleted share: %s", share.ID)
+}
+
+// PrintShare prints some information of the share
+func PrintShare(t *testing.T, share *shares.Share) {
+	asJSON, err := json.MarshalIndent(share, "", " ")
```

Comment:
> Does the pretty-printing work around this issue?

It does, yes. I initially tried the same method you're using (`json.MarshalIndent`), but noticed it changes the names to JSON format. That's when I went hunting for an external library. It'd be great if printing a nicely formatted struct was possible without an external dependency.

> I think printing the responses the way they are currently printed in many acceptance tests is a little cumbersome.

I agree 100%. When I initially added all of the `Print<Resource>` functions, it was easier for me to just manually inventory all of the fields. I figured I'd defer a better solution for later.

Edit: Additionally, if an external dependency is the only way to go, `kr/pretty` might not be the best choice. I also looked at `go-spew`, but that seemed to be _very_ verbose to the point where too much meta information got in the way.

I don't have a strong opinion about a solution, only that I agree with you that it'd be more maintainable to not manually inventory each field of the resource. :)